centreon-open-tickets/widgets/open-tickets/src/index.php

```diff
@@ -296,23 +324,84 @@
     }
 }
 
+if (! empty($preferences['poller'])) {
+    $mainQueryParameters[] = [
+        'parameter' => ':instance_id',
+        'value' => $preferences['poller'],
+        'type' => PDO::PARAM_INT
+    ];
+    $instanceIdCondition = ' i.instance_id = :instance_id';
+    $query = CentreonUtils::conditionBuilder($query, $instanceIdCondition);
+}
+
+if (! empty($preferences['poller'])) {
+    $resultsPoller = explode(',', $preferences['poller']);
+    $queryPoller = '';
+
+    foreach ($resultsPoller as $resultPoller) {
+        if ($queryPoller !== '') {
+            $queryPoller .= ', ';
+        }
+        $queryPoller .= ':instance_id_' . $resultPoller;
+
+        $mainQueryParameters[] = [
+            'parameter' => ':instance_id_' . $resultPoller,
+            'value' => (int)$resultPoller,
+            'type' => PDO::PARAM_INT
+        ];
+    }
+    $instanceIdCondition = ' i.instance_id IN (' . $queryPoller . ')';
```

Comment:
```suggestion
    $instanceIdCondition = ' h.instance_id IN (' . $queryPoller . ')';
```

there's no need to go into the instances table to filter on the instance id

Reply:
Fixed ^^